### PR TITLE
Sum All Integers in a String

### PR DESCRIPTION
**Pull request by ErdemT09**
Quesstion: Given a string str, return the sum of integers in this string (sign is also considered).

I used here the algorithm for parsing integers to string that we had used in [536. Construct Binary Tree from String](https://github.com/spiralgo/algorithms/blob/master/src/main/java/algorithms/curated170/medium/ConstructBinaryTreefromStringRecursive.java) and  [604. Design Compressed String Iterator](https://github.com/spiralgo/algorithms/blob/master/src/main/java/algorithms/curated170/easy/DesignCompressedStringIterator.java). This problem and the solution here is another demonstration of this algorithm, good as a simpler example.

### CODE REVIEW

File: src/main/java/algorithms/SumAllIntegersInString.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+public class SumAllIntegersInString {
+    public static int sumAllIntsInString0(String str) {
+        int sum = 0;
+        int currNum = 0;
+        int sign = 1;
+        for (int i = 0; i < str.length(); i++) {
+            char ch = str.charAt(i);
+            if (ch == '-') {
+                if (currNum != 0) {
```
Comment (altayhunoglu) [inline src/main/java/algorithms/SumAllIntegersInString.java:11]: We may not need this if-statement in this case.

Comment (ErdemT09) [reply src/main/java/algorithms/SumAllIntegersInString.java:11]: By line 11 or line 10? If it's line 11, we can indeed get rid of it.

Comment (altayhunoglu) [reply src/main/java/algorithms/SumAllIntegersInString.java:11]: Yeah, it is 11; as, even if 'currNum' is '0' it will not affect.

Comment (ErdemT09) [reply src/main/java/algorithms/SumAllIntegersInString.java]: Updated.

**Review by altayhunoglu — APPROVED** [review]
Quite clean code. Thanks.

**Pull request merged**